### PR TITLE
SyMS: Fix SDK breaking change [2023]

### PR DESCRIPTION
**Pull request by piyushdubey**
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

File: specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json [hunk 1]
```diff
@@ -2189,7 +2189,7 @@
       "type": "string",
       "in": "header",
       "required": true,
-      "x-ms-parameter-location": "client",
+      "x-ms-parameter-location": "method",
```
Comment (weidongxu-microsoft) [inline specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json:2192]: I think SDK don't want this parameter at all. All SDK has built-in policy on `user-agent`.

However, I won't be able to merge this.

Comment (MaryGao) [reply specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json:2192]: Agreed with weidong, for just we would always add the user-agent header by default. If you want to custom it we have a client-level option to set prefix.

```
{
userAgentOptions: { userAgentPrefix: "AdvancedSample V1.0.0" }, // Customized user-agent string
}

```

Comment (weidongxu-microsoft) [reply specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json:2192]: All SDK has user-agent built-in. It takes below form by default:
https://azure.github.io/azure-sdk/general_azurecore.html#telemetry-policy

PS: even `curl` sets the user-agent. So, just don't put this to swagger...

Comment (weidongxu-microsoft) [reply specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json:2192]: I may not get what you mean by "suffix route wouldn't work". Synapse **requires** a certain user-agent from user to use the API?

Comment (piyushdubey) [reply specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json:2192]: What I meant by suffix route not working is that we will to make a code change in the service to process the header with a suffix. However, we wouldn't need this as I have removed the user-agent param completely from the swagger. See below


Removed the user-agent param entirely from the Swagger - see this [PR](https://github.com/Azure/azure-rest-api-specs/pull/24469)

Resolving this conversation.

**Comment by piyushdubey** [comment]
UserAgent param is automatically passed with the API calls. Hence the change in this PR is not needed.

**Pull request closed without merging**